In [15]:
import numpy as np
import os
import pandas as pd
import sys
%matplotlib inline

In [2]:
import geopandas as gpd

In [13]:
!cd ~; cd yushi; cd USA/project2/shp/idw_res; ls

reg_res_idw.CPG  reg_res_idw.sbn  reg_res_idw.shp.NV-NYU-RDS2.5636.9808.sr.lock
reg_res_idw.dbf  reg_res_idw.sbx  reg_res_idw.shp.xml
reg_res_idw.prj  reg_res_idw.shp  reg_res_idw.shx


In [17]:
shp  = gpd.read_file('/nfshome/yc3763/yushi/USA/project2/shp/idw_res/reg_res_idw.shp')
shp.head()

Measured   Predicted       Error  Source_ID Included  Shape_Leng  \
0     108.0   69.528129  -38.471871          1      Yes    0.163067   
1     105.0   90.808291  -14.191709          2      Yes    0.347990   
2      66.0  103.500425   37.500425          3      Yes    0.964305   
3     104.0   76.524169  -27.475831          4      Yes    0.153547   
4     662.0  242.410285 -419.589715          5      Yes    0.154277   

   Shape_Area                                           geometry  
0    0.000972  POLYGON ((-73.89641133483127 40.9045045208203,...  
1    0.001876  POLYGON ((-73.84423146899854 40.86942035096848...  
2    0.002649  (POLYGON ((-73.78833349834531 40.8346671297594...  
3    0.001054  POLYGON ((-73.84792614069238 40.87134223307794...  
4    0.000802  POLYGON ((-73.95374853778469 40.63858633758554...

In [31]:
url = "https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt"     

Question: is the new Bus route improving commute?

H0: the commute time is on average the same or longer with the new bus route as it was before: Tnew >= Told

Ha: the commute time is shorter with the new bus route than it was before: Tnew < Told

I set the significance level as: $\alpha=0.05$

In [32]:
bus = pd.read_csv(url, sep=",", names = ["Commute Time"])

In [33]:
bus.head()

Commute Time
0     31.622239
1     32.821376
2     30.229101
3     31.413766
4     39.010550

In [37]:
x = bus['Commute Time'].mean()
y = bus['Commute Time'].std()
N = len(bus)
print ("Mean:", x, "\nStandard Deviation:", y)

Mean: 34.4661616883 
Standard Deviation: 7.13728013375


In [38]:
pmean = 36
pstd = 6

In [39]:
# Calculating Z score 
a = pmean - x 
b = pstd / np.sqrt(N)
zscore= a/b
print(zscore)

2.55639718617


The Z-test score is 2.556, which is higher than the significant level 0.05, correspondence to 2 signma, as i setted before, thus the null hypothsis can be rejected. It also means that the new bus route is effecient on commuting time. 